File: Application/Backend/PlanetWars/DTOs/SessionDto.cs
using System;
using System.Collections.Generic;

namespace PlanetWars.DTOs
{
    public class SessionDto
    {
        public Guid ID { get; set; }

        public string Name { get; set; }

        public string GameCode { get; set; }

        public int CurrentTurnIndex { get; set; }

        public List<PlayerDto> Players { get; set; }
        
        public int PlayerCount { get; set; }

        public GalaxyDto Galaxy { get; set; }

        public int MaxPlayers { get; set; }
    }
}

File: Application/Frontend/planet-wars/src/app/feature/game/dtos/sessionDto.ts
import { GalaxyDto } from './galaxyDto';
import { PlayerDto } from './playerDto';

export interface SessionDto {
  id: string;
  name: string;
  gameCode: string;
  currentTurnIndex: number;
  players: PlayerDto[];
  playerCount: number;
  galaxy: GalaxyDto;
  maxPlayers: number;
}
